### ISSUE TITLE
Generation of RW rules from annotations

### ISSUE THREAD

**Issue by aslesarenko** [scalan/scalan#96]
If implement `Arrays` as wrappers some RW rules will not be generated.

this should be fixed by generating rules from annotations.

See how rules for `Seq` are generated now

**Comment by alexeyr** [comment]
Note: additional rewrite rules can always be written in a separate cake component.

**Comment by aslesarenko** [comment]
@alexeyr yes, but this issue is about automatic generation so that manual writing is not required. They can be generated in separate cake, though :)
